model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
+	colon := strings.Index(hostAddr, ":")
+	if colon < 0 {
+		return fmt.Errorf("':' separator not found in %q, host address must be of the form <DNS name>:<port> or <IP>:<port>",
+			hostAddr)
+	}
+	port, err := strconv.Atoi(hostAddr[colon+1:])
+	if err != nil {
+		return err
+	}
+
+	if err = validatePort(port); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateDuration(pd *duration.Duration) error {
+	dur, err := ptypes.Duration(pd)
+	if err != nil {
+		return err
+	}
+	if dur <= 0 {
```

Comment:
Maybe a good idea to check max duration here as well? This would need to be specified as paramter to this function but would help catch fat-fingering Xs instead of Xs when hand editing config maps.